sentry-android-core/src/main/java/io/sentry/android/core/AndroidTransactionProfiler.java

```diff
@@ -138,9 +138,13 @@ private void init() {
     traceFilesDir = new File(tracesFilesDirPath);
   }
 
-  @SuppressLint("NewApi")
   @Override
   public synchronized void onTransactionStart(@NotNull ITransaction transaction) {
+    options.getExecutorService().submit(() -> onTransactionStartSafe(transaction));
```

Comment:
We can either use the `options.getExecutorService().submit` or simply call the method directly.
The only remaining io method is `traceFilesDir.canWrite()` in `onTransactionStartSafe`
Is there any concern in using the executorService?

Reply:
I think it should be fine, since we're using a single thread executor, so there should be no conflicts